src/EnergyPlus/WaterThermalTanks.cc

```diff
@@ -3641,128 +3642,176 @@ bool getWaterTankMixedInput(EnergyPlusData &state)
+                ErrorsFound = true;
+            }
+            Tank.HeaterHeight1 = state.dataInputProcessing->inputProcessor->getRealFieldValue(fields, schemaProps, "top_temperature_sensor_height");
+            Tank.TempSensorHeight1 = Tank.HeaterHeight1;
+            Tank.HeaterHeight2 =
+                state.dataInputProcessing->inputProcessor->getRealFieldValue(fields, schemaProps, "bottom_temperature_sensor_height");
+            Tank.TempSensorHeight2 = Tank.HeaterHeight2;
+            // use side
+            auto const &UseFlowDirectionScheName = fields.find("use_side_flow_direction_schedule");
+            if (UseFlowDirectionScheName == fields.end() || UseFlowDirectionScheName->empty()) {
+                Tank.UseFlowDirectionSched = Sched::GetScheduleAlwaysOn(state);
+            } else if ((Tank.UseFlowDirectionSched =
+                            Sched::GetSchedule(state, Util::makeUPPER(UseFlowDirectionScheName.value().get<std::string>()))) == nullptr) {
+                ShowSevereItemNotFound(state, eoh, "use_side_flow_direction_schedule", UseFlowDirectionScheName.value().get<std::string>());
+                ErrorsFound = true;
+            }
+            kwHeatingCooling = "heating";
         }
 
-        if (state.dataIPShortCut->lAlphaFieldBlanks(3)) {
-            ShowSevereEmptyField(state, eoh, state.dataIPShortCut->cAlphaFieldNames(3));
-            ErrorsFound = true;
-        } else if ((Tank.setptTempSched = Sched::GetSchedule(state, state.dataIPShortCut->cAlphaArgs(3))) == nullptr) {
-            ShowSevereItemNotFound(state, eoh, state.dataIPShortCut->cAlphaFieldNames(3), state.dataIPShortCut->cAlphaArgs(3));
-            ErrorsFound = true;
+        if (TankTempLimit > 0.0) {
+            Tank.TankTempLimit = TankTempLimit;
+        } else {
+            Tank.TankTempLimit = 1.0;
         }
 
-        if (state.dataIPShortCut->rNumericArgs(4) > 0.0) {
-            Tank.DeadBandDeltaTemp = state.dataIPShortCut->rNumericArgs(4);
+        auto const DeadBandDeltaTemp =
+            state.dataInputProcessing->inputProcessor->getRealFieldValue(fields, schemaProps, "deadband_temperature_difference");
+        if (DeadBandDeltaTemp > 0.0) {
+            Tank.DeadBandDeltaTemp = DeadBandDeltaTemp;
         } else {
             // Default to very small number (however it can't be TINY or it will break the algorithm)
             Tank.DeadBandDeltaTemp = 0.0001;
         }
 
-        Tank.HeaterHeight1 = state.dataIPShortCut->rNumericArgs(5);
-        Tank.MaxCapacity = state.dataIPShortCut->rNumericArgs(7);
-        if (Tank.MaxCapacity == DataSizing::AutoSize) {
+        auto const &MaxCapacity = fields.at(format("nominal_{}_capacity", kwHeatingCooling));
+        if (MaxCapacity == "Autosize") {
+            Tank.MaxCapacity = DataSizing::AutoSize;
             Tank.MaxCapacityWasAutoSized = true;
+        } else {
+            Tank.MaxCapacity = MaxCapacity.get<Real64>();
         }
```

Comment:
Would this be a better pattern? And why use format when you know what the field name is, and it's the only one?

    Tank.MaxCapacity = state.dataInputProcessing->inputProcessor->getRealFieldValue(chiller_fields, schemaProps, "nominal_heating_capacity");
    if (Tank.MaxCapacity == DataSizing::AutoSize) {
        Tank.MaxCapacityWasAutoSized = true;
    }

Reply:
This is better. I'll change to using this pattern